front/app/components/admin/PostManager/CommonGroundInputManager/index.tsx

```diff
@@ -0,0 +1,277 @@
+  NoPostDescription,
+} from 'components/admin/PostManager/components/PostTable/NoPost';
+import { TitleLink } from 'components/admin/PostManager/components/PostTable/Row';
+import Pagination from 'components/Pagination';
+import T from 'components/T';
+import Button from 'components/UI/ButtonWithLink';
+import Warning from 'components/UI/Warning';
+
+import { FormattedMessage } from 'utils/cl-intl';
+import { getPageNumberFromUrl } from 'utils/paginationUtils';
+
+import messages from './messages';
+
+interface Props {
+  projectId: string;
+  phaseId: string;
+  setShowPastInputsModal: (value: React.SetStateAction<boolean>) => void;
+}
+
+const CommonGroundInputManager = ({
+  projectId,
+  phaseId,
+  setShowPastInputsModal,
+}: Props) => {
+  const { data: project } = useProjectById(projectId);
+  const [queryParameters, setQueryParameters] = useState<IIdeaQueryParameters>({
+    'page[size]': 10,
+    'page[number]': 1,
+    sort: 'new',
+    projects: [projectId],
+    phase: phaseId,
+    transitive: false,
+  });
+
+  const [previewPostId, setPreviewPostId] = useState<string | null>(null);
+  const [previewMode, setPreviewMode] = useState<PreviewMode>('view');
+  const [selection, setSelection] = useState<Set<string>>(new Set());
+  const { data: jobProgressData } = useJobInProgressByPhase(phaseId);
+  const { progress: importedCount, total: totalInputsToBeImported } =
+    jobProgressData?.data[0]?.attributes || {};
+  const isJobInProgress = importedCount !== undefined;
+
+  const resetSelection = () => setSelection(new Set());
+
+  useEffect(() => {
+    setQueryParameters((prevParams) => ({
+      ...prevParams,
+      'page[number]': 1,
+      projects: [projectId],
+      phase: phaseId,
+    }));
+
+    resetSelection();
+  }, [projectId, phaseId]);
+
+  const { data: ideas, isLoading } = useIdeas(queryParameters);
+
+  const openPreviewEdit = () => {
+    const id = [...selection][0];
+    if (selection.size === 1 && id) {
+      setPreviewMode('edit');
+      setPreviewPostId(id);
+    }
+  };
+
+  const openPreview = (id: string) => {
+    setPreviewPostId(id);
+    setPreviewMode('view');
+  };
+  const closePreview = () => {
+    setPreviewPostId(null);
+    setPreviewMode('view');
+  };
+
+  const allSelected = !!(
+    ideas &&
+    ideas.data.length > 0 &&
+    ideas.data.every((i) => selection.has(i.id))
+  );
+  const toggleSelectAll = () => {
+    if (allSelected) setSelection(new Set());
+    else if (ideas) setSelection(new Set(ideas.data.map((i) => i.id)));
+  };
+  const toggleSelect = (id: string) => {
+    setSelection((prev) => {
+      const next = new Set(prev);
+      next.has(id) ? next.delete(id) : next.add(id);
+      return next;
+    });
+  };
+
+  if (isLoading) {
+    return (
+      <Box display="flex" justifyContent="center" p="20px">
+        <Spinner />
+      </Box>
+    );
+  }
+
+  if (!ideas) {
+    return null;
+  }
+
+  if (ideas.data.length === 0) {
+    return (
+      <NoPostPage>
```

Comment:
This file is a bit long - I wonder if some of this could be separated into smaller components? 

Not blocking though :)